cli/src/cmd/forge/create.rs

```diff
@@ -115,11 +115,18 @@ impl CreateArgs {
             Some(ref v) => {
                 let constructor_args =
                     if let Some(ref constructor_args_path) = self.constructor_args_path {
-                        if !std::path::Path::new(&constructor_args_path).exists() {
+                        let path = std::path::Path::new(&constructor_args_path);
+                        if !path.exists() {
                             eyre::bail!("constructor args path not found");
                         }
-                        let file = fs::read_to_string(constructor_args_path)?;
-                        file.split(' ').map(|s| s.to_string()).collect::<Vec<String>>()
+                        if path.extension() == Some(std::ffi::OsStr::new("json")) {
+                            let file = fs::File::open(constructor_args_path)?;
+                            serde_json::from_reader(file)
+                                .expect("constructor args file content should be JSON array")
+                        } else {
```

Comment:
there's `ethers::solc::utils::read_json_file`

Reply:
I have done this now. I also handle the error a bit better so that it uses eyre::bail and not panic when the json file format is incorrect.